src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareIncludeExcludeNodeOperation.cs

```diff
@@ -97,10 +142,14 @@ private bool IsEqual(SchemaDifference difference, DiffEntry diffEntry)
             System.Reflection.PropertyInfo[] properties = diffEntry.GetType().GetProperties();
             foreach (var prop in properties)
             {
-                result = result &&
-                    ((prop.GetValue(diffEntry) == null &&
-                    prop.GetValue(entryFromDifference) == null) ||
-                    prop.GetValue(diffEntry).SafeToString().Equals(prop.GetValue(entryFromDifference).SafeToString()));
+                // Don't need to check if included is the same when verifying if the difference is equal
+                if (prop.Name != "Included")
+                {
+                    result = result &&
```

Comment:
>result [](start = 29, length = 6)

Seems like we can optimize this by returning as soon as result is false, rather than continuing to iterate over the foreach loop.

Reply:
reflection is SLOW if you didn't ready know

---
In reply to: [338769969](https://github.com/microsoft/sqltoolsservice/pull/881#discussion_r338769969) [](ancestors = 338769969)